src/wallet/rpcwallet.cpp

```diff
@@ -2004,7 +1997,14 @@ static UniValue walletpassphrase(const JSONRPCRequest& request)
     pwallet->TopUpKeyPool();
 
     pwallet->nRelockTime = GetTime() + nSleepTime;
-    RPCRunLater(strprintf("lockwallet(%s)", pwallet->GetName()), std::bind(LockWallet, pwallet), nSleepTime);
+    std::weak_ptr<CWallet> weak_wallet = wallet;
+    RPCRunLater(strprintf("lockwallet(%s)", pwallet->GetName()), [weak_wallet] {
+        if (auto wallet = weak_wallet.lock()) {
```

Comment:
Use another name here to avoid shadowing existing local variable `wallet` :-)

Reply:
`wallet` is not captured.

Reply:
Sorry, sloppy wording from me - it is not a local variable. Still better to minimize surprises by not re-using the variable name, no?

Reply:
What surprise?

Reply:
@practicalswift renamed.